### PR TITLE
 Remove attach of bearer token when bucket owner is not an issuer of the bearer token  

### LINKED ISSUE #459: Fix `test_object_copy_canned_acl`
This test sets canned ACL to the object. 

```py
def test_object_copy_canned_acl():
    bucket_name = get_new_bucket()
    client = get_client()
    alt_client = get_alt_client()
    client.put_object(Bucket=bucket_name, Key='foo123bar', Body='foo')

    copy_source = {'Bucket': bucket_name, 'Key': 'foo123bar'}
    client.copy_object(Bucket=bucket_name, CopySource=copy_source, Key='bar321foo', ACL='public-read')
    ## check ACL is applied by doing GET from another user
    alt_client.get_object(Bucket=bucket_name, Key='bar321foo')  # <- FAILED HERE

    metadata={'abc': 'def'}
    copy_source = {'Bucket': bucket_name, 'Key': 'bar321foo'}
    client.copy_object(ACL='public-read', Bucket=bucket_name, CopySource=copy_source, Key='foo123bar', Metadata=metadata, MetadataDirective='REPLACE')

    # check ACL is applied by doing GET from another user
    alt_client.get_object(Bucket=bucket_name, Key='foo123bar')
```

ACL is not processed during object copying, therefore alt client fails with 
```
layer/object.go:710     error was transformed   {"request_id": "46f7588e-02a7-47ff-94ff-6d26354983bd", "error": "access denied: access to operation SEARCH is denied by extended ACL check: bearer token is not signed by the container owner"}
handler/util.go:25      could not find object   {"request_id": "46f7588e-02a7-47ff-94ff-6d26354983bd", "method": "GetObject", "bucket_name": "yournamehere-sdcore7ma75qo04d-1", "object_name": "bar321foo", "error": "AccessDenied: 403 => Access Denied."}
```

### PR DESCRIPTION
**Pull request by masterSplinter01**
The `test_object_copy_canned_acl` test still not passing because `main_wallet` creates a private container in tests and `alt_wallet` can't `get-object` from private container, even the object was put with `public-read` acl. 

I tried to make `main-wallet` create `public-read` container and the test began to pass.   

I suggest to fix the test and place it in `test_s3_neofs.py`. 

Closes #459

### CODE REVIEW

File: api/handler/get.go [hunk 1]
```diff
@@ -168,7 +168,6 @@ func (h *handler) GetObjectHandler(w http.ResponseWriter, r *http.Request) {
 		ObjectInfo: info,
 		Writer:     w,
 		Range:      params,
-		VersionID:  p.VersionID,
```
Comment (masterSplinter01) [inline api/handler/get.go:171]: It seems like the field is not used, also we have `versionID` in `ObjectInfo` field

File: api/layer/layer.go [hunk 2]
```diff
@@ -323,10 +322,12 @@ func (n *layer) Owner(ctx context.Context) user.ID {
 	return ownerID
 }
 
-func (n *layer) prepareAuthParameters(ctx context.Context, prm *neofs.PrmAuth) {
+func (n *layer) prepareAuthParameters(ctx context.Context, prm *neofs.PrmAuth, bktOwner user.ID) {
 	if bd, ok := ctx.Value(api.BoxData).(*accessbox.Box); ok && bd != nil && bd.Gate != nil {
-		prm.BearerToken = bd.Gate.BearerToken
-		return
+		if issuer, ok := bd.Gate.BearerToken.Issuer(); ok && bktOwner.Equals(issuer) {
```
Comment (KirillovDenis) [inline api/layer/layer.go:327]: It seems we can adopt https://github.com/nspcc-dev/neofs-s3-gw/pull/485

File: api/layer/object.go [hunk 3]
```diff
@@ -437,23 +444,22 @@ func (n *layer) headVersion(ctx context.Context, bkt *data.BucketInfo, p *HeadOb
...
-	n.objCache.Delete(newAddress(idCnr, idObj))
+	n.objCache.Delete(newAddress(bktInfo.CID, idObj))
 
 	return n.transformNeofsError(ctx, n.neoFS.DeleteObject(ctx, prm))
 }
 
 // objectPutAndHash prepare auth parameters and invoke neofs.CreateObject.
 // Returns object ID and payload sha256 hash.
 func (n *layer) objectPutAndHash(ctx context.Context, prm neofs.PrmObjectCreate) (*oid.ID, []byte, error) {
-	n.prepareAuthParameters(ctx, &prm.PrmAuth)
```
Comment (KirillovDenis) [inline api/layer/object.go:456]: Maybe we can provide `bktInfo` to this function to preserve invocation of the `n.prepareAuthParameters` here?

File: api/layer/layer.go [hunk 4]
```diff
@@ -380,8 +381,7 @@ func (n *layer) ListBuckets(ctx context.Context) ([]*data.BucketInfo, error) {
 func (n *layer) GetObject(ctx context.Context, p *GetObjectParams) error {
 	var params getParams
 
-	params.oid = p.ObjectInfo.ID
-	params.cid = p.ObjectInfo.CID
+	params.objInfo = p.ObjectInfo
```
Comment (KirillovDenis) [inline api/layer/layer.go:384]: I suppose we can keep the old `getParams` fields. And extend `GetObjectParams` by `bktInfo` and pass it (`bktInfo`) as the second parameter to `n.initObjectPayloadReader` function

**Comment by KirillovDenis** [comment]
Please fix tests

File: api/layer/layer.go (hunk 2 above)
Comment (alexvanin) [reply api/layer/layer.go:327]: I would like to rebase `tree-service` branch after this PR. Adopt it in separate PR.

**Comment by alexvanin** [comment]
> can't get-object from private container, even the object was put with public-read acl.
>
> I suggest to fix the test and place it in test_s3_neofs.py.

It's more like new S3 issue to fix object ACLs, isn't it?

1. `main_wallet` creates private container -- extended ACL denied for everyone
2. `main_wallet` sets public read acl for the object -- extended ACL has one more additional `allow` rule to this object for target `OTHER`, it is a first rule in the table
3. `alt_wallet` tries to get object -- without bearer token NeoFS node applies container EACL to request => first rule allows to get object.

File: api/layer/layer.go (hunk 4 above)
Comment (masterSplinter01) [reply api/layer/layer.go:386]: Removed this with dragging of `bktInfo` through params. Added a separated commit to make it easier to view changes

**Review by KirillovDenis — APPROVED** [review]

**Comment by alexvanin** [comment]
> > can't get-object from private container, even the object was put with public-read acl.
> > I suggest to fix the test and place it in test_s3_neofs.py.
> 
> It's more like new S3 issue to fix object ACLs, isn't it?
> 
>     1. `main_wallet` creates private container -- extended ACL with 7 `deny` rules for target `OTHER`
> 
>     2. `main_wallet` sets public read acl for the object -- extended ACL has one more additional `allow` rule to this object for target `OTHER`, it is a first rule in the table
> 
>     3. `alt_wallet` tries to get object -- without bearer token NeoFS node applies container EACL to request => first rule allows to get object.

As @masterSplinter01 mentioned, we got access deny error because of `search` request. Which is fair enough, because we do search in the bucket. `tree-service` branch avoids extra search, so maybe the issue should be gone. Let's investigate that and fix per object public ACL in private containers.

File: api/layer/layer.go (hunk 2 above)
Comment (alexvanin) [reply api/layer/layer.go:329]: I messed up with the branches. Both SDK update and this PR were into `master` branch, so we could adopt it. For some reason I thought this PR has `tree-service` target.

I pushed https://github.com/nspcc-dev/neofs-s3-gw/commit/4ed939773b020ee6e305f9123b53e62cdbd6f859 into master branch.

**Pull request merged**